client/allocrunner/taskrunner/task_runner.go

```diff
@@ -988,7 +993,7 @@ func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
 		Resources: &drivers.Resources{
 			NomadResources: taskResources,
 			LinuxResources: &drivers.LinuxResources{
-				MemoryLimitBytes: taskResources.Memory.MemoryMB * 1024 * 1024,
+				MemoryLimitBytes: memoryLimit * 1024 * 1024,
```

Comment:
Updating the linux resources is intended to ease drivers implementation and adoption of the features: drivers that use `resources.LinuxResources.MemoryLimitBytes` don't need to be updated.

Drivers that use NomadResources will need to updated to track the new field value. Given that tasks aren't guaranteed to use up the excess memory limit, this is a reasonable compromise.

I don't know the breakdown of how external 3rd party drivers check memory limit, but happy to change the default.

Reply:
> Drivers that use NomadResources will need to updated to track the new field value. Given that tasks aren't guaranteed to use up the excess memory limit, this is a reasonable compromise.

So if they don't get updated, they'll just end up setting their limit equal to the `memory` field value, just as they do today? They just end up ignoring `memory_max`? 

From a customer/community impact standpoint, the two I'd worry the most about are [`containerd`](https://github.com/Roblox/nomad-driver-containerd/blob/v0.8/containerd/driver.go#L434) and [`podman`](https://github.com/hashicorp/nomad-driver-podman/blob/v0.2.0/driver.go#L389-L392). Also, do we want to update [`qemu`](https://github.com/hashicorp/nomad/blob/v1.0.4/drivers/qemu/driver.go#L376) to take whichever is greater?

Reply:
Yes, the failure mode is ignoring memory_max and behaving like today.  I'm researching soft vs hard limits a bit now, and will ensure `containerd` and `podman` are updated to the recommended pattern.